exonum/src/crypto/crypto_lib/sodiumoxide/x25519.rs

```diff
@@ -137,7 +135,7 @@ macro_rules! implement_x25519_type {
         fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
             write!(f, stringify!($name))?;
             write!(f, "(")?;
-            for i in &self.0[0..BYTES_IN_DEBUG] {
+            for i in &self.0[..] {
```

Comment:
`[..]` is redundant here

Reply:
I remembered why I did this in this way: using `self.0` and `self` did not auto-convert to iterable slices for some types declared by these macros (while some others were fine).

I am going to write a single utility function for hex-formatting slices with ellipses and will do something about this redundancy.

Reply:
Okay... Here's the catch with [ed25519's types that we're using][1]. All types implement `Index<Output=[u8]>`, but only 'public' types (PublicKey and Signature) implement `AsRef<[u8]>`. So at least for 'private' types we'll have to use `&self[..]` or something to first convert them into a slice via range-indexing. Taking a reference works only for 'public' types.

[1]: https://sodiumoxide.github.io/sodiumoxide/sodiumoxide/crypto/sign/ed25519/index.html